src/app/speakers/speaker.models.ts

```diff
@@ -0,0 +1,25 @@
+export class Speaker {
+  public name: string = 'Jhon Doe';
+  public title: string = 'Google Developer';
+  public bio: string = 'lorem ipsum dolor sit amet, lorem ipsum dolor sit amet, lorem ipsum dolor sit amet.';
+  public photo: string = null;
+  public twitter: string = null;
+  public github: string = null;
+
+  constructor(speaker: Speaker = null) {
```

Comment:
How about if we use the shortcut syntax?

`constructor(public name: string, ...)`

Also, comparison with `null` are unnecessary in this case. You can simply put: `if (!this.photo) {`

Reply:
For the first one the issue is the order matters for the parameters, there are discussions about named params being better, i personally think it's better because i can override any of the defaults without having to declare each other parameter, though if you want it back to shortcut syntax it's fine by me.

I wanted to avoid coercion of variable values, `!this.photo` would yield false both if `null`, empty string or any falsy value.

Reply:
On one side we could have unordered parameters (or even null) but at the cost of having to:
  1. Define the variables and assign their value in the class instead of using the shortcut syntax
  2. Having to write the properties at the time of creating an instance making it more verbose to do it
  3. Not being able to set default parameters for null variables (Which would remove the condition for the photo on line 21 by using the `=` in the constructor)

Reply:
1. True, would be more verbose.
2. It's safer to name the variables that goes into a function, that's critical if there are many, so we don't rely on order (i think i read somewhere that this pattern is better for >= 5 arguments).
3. We can set default parameters either by using the defaults above in the variable definition, or through destructuring in the constructor parameters, the photo one is special because i need a random number for a random photo in case no photo is passed as param.

Reply:
Do whatever the fuck you want

Reply:
Kidding. Given that this project is simple and no one else will instantiate speakers more than the speaker component, I'd just leave it with the shortcut syntax. So it's less verbose. For for sophitisticated apps I'd go with your approach though 👍

Reply:
Yes Sr.